src/ray/core_worker/transport/actor_scheduling_queue.cc

```diff
@@ -146,26 +162,38 @@ void ActorSchedulingQueue::ScheduleRequests() {
     pending_actor_tasks_.erase(head);
   }
 
+  // Process as many retry requests as we can.
+  auto retry_iter = pending_retry_actor_tasks_.begin();
+  while (retry_iter != pending_retry_actor_tasks_.end()) {
```

Comment:
iteration order here is non-deterministic, would be better to execute them in FIFO order IMO, otherwise you could have starvation issues

Reply:
a queue logically makes more sense for the pending retry actor tasks given they don't respect the seq no anyways